### PR TITLE
feat: upload client logs while stopping the network

### PR DESCRIPTION
**Pull request by RolandSherwin**
- Tar and upload `client` and `heapnode` logs if they exist.

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -176,40 +176,64 @@ runs:
 
     - name: Tar log files (Linux)
       if: inputs.platform == 'ubuntu-latest' && inputs.action == 'stop'
+      env:
+        ROOT_DIR: "/home/runner/.local/share/safe/"
```
Comment (jacderida) [inline action.yml:180]: Hi Roland,

Just a style issue. I would prefer if the trailing slash could be removed from these variables.

I personally think that
```
find "${ROOT_DIR}/node" -iname '*.log*' | tar -zcvf node_log_files.tar.gz --files-from -
```
Is more readable than
```
find "${ROOT_DIR}node" -iname '*.log*' | tar -zcvf node_log_files.tar.gz --files-from -
```

The same would apply to the other relevant variables.

File: action.yml [hunk 2]
```diff
@@ -176,40 +176,64 @@ runs:
...
+            Set-Location -Path $destDirectory
+            7z a -ttar 'logs.tar' .\temp\*
+            7z a -tgzip $archiveName 'logs.tar'
+
+            Remove-Item -Recurse -Force 'temp'
+            Remove-Item 'logs.tar'
+        }
+        
+        # Creating a single tarball with both client and node logs
+        Get-ChildItem . -Filter *_log_files.tar.gz | % {7z a -ttar 'logs.tar' $_.FullName}
```
Comment (jacderida) [inline action.yml:232]: Just wanted to check, did you do a run and see this working ok?

File: action.yml (hunk 1 above)
Comment (RolandSherwin) [reply action.yml]: Yeah, that makes it even more readable. Updated my PR.

File: action.yml (hunk 2 above)
Comment (RolandSherwin) [reply action.yml]: Yeah! I've run the script on my Windows machine and everything works fine.
Though every `.tar.gz` file contains another `.tar` which then contains the logs. But reading about it, I think there is no workaround for this?

Anyway, I've updated the PR to make this line more readable.

**Pull request merged**